### ISSUE TITLE
Crash with group windows

### ISSUE THREAD

**Issue by apppppppple** [hyprwm/Hyprland#308]
Hyprland will crash when using group windows in certain ways, one I've noticed is when adding a new window to the group after the group was formed, and then pressing the bind to create a new group while the window is focused.

Repro steps:

1. Open Hyprland
2. Open a terminal
3. Run this script:
```sh
hyprctl dispatch exec foot && sleep 1 && hyprctl dispatch exec foot && sleep 1 && hyprctl dispatch togglegroup '' && sleep 1 && hyprctl dispatch movefocus r && sleep 1 && hyprctl dispatch togglegroup '' && sleep 1 && hyprctl dispatch exec foot && sleep 1 && hyprctl dispatch togglegroup ''
```
(you can most likely replace foot with anything else that opens a window, it's just my te)
4. When it finishes, hyprland will freeze for several seconds and then crash

Stack trace:
```
           PID: 938983 (Hyprland)
           UID: 1000 (apple)
           GID: 100 (users)
        Signal: 11 (SEGV)
     Timestamp: Tue 2022-07-05 11:15:55 EDT (42s ago)
  Command Line: Hyprland
    Executable: /nix/store/xswp44cblaskgywj3gfmnqlk3qargkx9-hyprland-debug-0.6.2beta+date=2022-07-05/bin/Hyprland
 Control Group: /user.slice/user-1000.slice/session-4.scope
          Unit: session-4.scope
         Slice: user-1000.slice
       Session: 4
     Owner UID: 1000 (apple)
       Boot ID: af3383fc434140d9bd286c638ea3a272
    Machine ID: b382ce9a516b438ebfdaa2c9bf114461
      Hostname: nixos
       Storage: /var/lib/systemd/coredump/core.Hyprland.1000.af3383fc434140d9bd286c638ea3a272.938983.1657034155000000.zst (truncated)
     Disk Size: 1.3M
       Message: Process 938983 (Hyprland) of user 1000 dumped core.
                
                Module /nix/store/xswp44cblaskgywj3gfmnqlk3qargkx9-hyprland-debug-0.6.2beta+date=2022-07-05/bin/Hyprland without build-id.
                Stack trace of thread 938983:
                #0  0x000000000047b9bf _ZN7CWindow17updateWindowDecosEv (/nix/store/xswp44cblaskgywj3gfmnqlk3qargkx9-hyprland-debug-0.6.2beta+date=2022-07-05/bin/Hyprland + 0x7b9bf)
                
                Stack trace of thread 938992:
                #0  0x00007f4e9f142cb5 n/a (n/a + 0x0)
                ELF object binary architecture: AMD x86-64
```
[crashhypr.log](https://github.com/hyprwm/Hyprland/files/9047409/crashhypr.log)
It's possible that a more minimal repro exists, but I haven't found one thus far.

**Comment by apppppppple** [comment]
Is it possible I attached the wrong core dump or log? I'm sure I never pressed the keybind, and I've run this several times

**Comment by apppppppple** [comment]
My apologies, it looks like I did attach the wrong one. This is the correct core dump:
```
           PID: 938983 (Hyprland)
           UID: 1000 (apple)
           GID: 100 (users)
        Signal: 11 (SEGV)
     Timestamp: Tue 2022-07-05 11:15:55 EDT (42s ago)
  Command Line: Hyprland
    Executable: /nix/store/xswp44cblaskgywj3gfmnqlk3qargkx9-hyprland-debug-0.6.2beta+date=2022-07-05/bin/Hyprland
 Control Group: /user.slice/user-1000.slice/session-4.scope
          Unit: session-4.scope
         Slice: user-1000.slice
       Session: 4
     Owner UID: 1000 (apple)
       Boot ID: af3383fc434140d9bd286c638ea3a272
    Machine ID: b382ce9a516b438ebfdaa2c9bf114461
      Hostname: nixos
       Storage: /var/lib/systemd/coredump/core.Hyprland.1000.af3383fc434140d9bd286c638ea3a272.938983.1657034155000000.zst (truncated)
     Disk Size: 1.3M
       Message: Process 938983 (Hyprland) of user 1000 dumped core.
                
                Module /nix/store/xswp44cblaskgywj3gfmnqlk3qargkx9-hyprland-debug-0.6.2beta+date=2022-07-05/bin/Hyprland without build-id.
                Stack trace of thread 938983:
                #0  0x000000000047b9bf _ZN7CWindow17updateWindowDecosEv (/nix/store/xswp44cblaskgywj3gfmnqlk3qargkx9-hyprland-debug-0.6.2beta+date=2022-07-05/bin/Hyprland + 0x7b9bf)
                
                Stack trace of thread 938992:
                #0  0x00007f4e9f142cb5 n/a (n/a + 0x0)
                ELF object binary architecture: AMD x86-64
```

**Comment by vaxerski** [comment]
a bit short innit

**Comment by apppppppple** [comment]
yeah :pensive: 
Anything else I could do? I switched to the debug build for this but it didn't seem to help lengthen it

**Comment by vaxerski** [comment]
I'll try to reproduce it, sec

**Comment by vaxerski** [comment]
ok got it

**Comment by vaxerski** [comment]
fixed in 6f5b8425cb7b7304dcce9a48e613d87538bd33d2
